io/io/src/TKey.cxx

```diff
@@ -353,6 +353,7 @@ TKey::TKey(const void *obj, const TClass *cl, const char *name, Int_t bufsize, T
          else               bufmax = kMAXZIPBUF;
          R__zipMultipleAlgorithm(cxlevel, &bufmax, objbuf, &bufmax, bufcur, &nout, cxAlgorithm);
          if (nout == 0 || nout >= fObjlen) { //this happens when the buffer cannot be compressed
+            delete[] fBuffer;
```

Comment:
This will invalidate `bufcur`; while it is not used anymore before returning, maybe it's better to explicitly set it to nullptr to avoid any future issue (the compiler will probably discard the operation in release so we're likely not paying any cost for added safety)